From a4a3b20f20909ef755139541b7d9609db8aa6ef0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maciej=20Kr=C3=BCger?= <mkg20001@gmail.com>
Date: Wed, 12 Feb 2025 20:16:44 +0100
Subject: [PATCH 1/2] ansible_2_16: fix build

the setuptools version restriction doesn't exist in 2_16
removed patch as it breaks the build
---
 pkgs/top-level/all-packages.nix | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 78a1272fc905..133f5159410b 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -7465,6 +7465,13 @@ with pkgs;
       inherit version;
       hash = "sha256-WeSqQO1azbTvm789BYkY//k/ZqFJNz2BWciilgRBC9o=";
     };
+
+    postPatch = ''
+      substituteInPlace lib/ansible/executor/task_executor.py \
+        --replace "[python," "["
+
+      patchShebangs --build packaging/cli-doc/build.py
+    '';
   }));
 
   ansible-builder = with python3Packages; toPythonApplication ansible-builder;
-- 
2.47.1

From f47cb5ab347006fa03bc87d8a76e242acbbcb73c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maciej=20Kr=C3=BCger?= <mkg20001@gmail.com>
Date: Wed, 12 Feb 2025 20:31:56 +0100
Subject: [PATCH 2/2] ansible_2_17: use correct version

---
 pkgs/top-level/all-packages.nix | 21 +++++++++++++++++++--
 1 file changed, 19 insertions(+), 2 deletions(-)

diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 133f5159410b..74a0806edebc 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -7457,8 +7457,25 @@ with pkgs;
 
   anybadge = with python3Packages; toPythonApplication anybadge;
 
-  ansible = ansible_2_17;
-  ansible_2_17 = python3Packages.toPythonApplication python3Packages.ansible-core;
+  ansible = ansible_2_18;
+  ansible_2_18 = python3Packages.toPythonApplication python3Packages.ansible-core;
+  ansible_2_17 = python3Packages.toPythonApplication (python3Packages.ansible-core.overridePythonAttrs (oldAttrs: rec {
+    version = "2.17.8";
+    src = oldAttrs.src.override {
+      inherit version;
+      hash = "sha256-Ob6KeYaix9NgabDZciC8L2eDxl/qfG1+Di0A0ayK+Hc=";
+    };
+
+    postPatch = ''
+      substituteInPlace lib/ansible/executor/task_executor.py \
+        --replace "[python," "["
+
+      patchShebangs --build packaging/cli-doc/build.py
+
+      substituteInPlace pyproject.toml \
+        --replace-fail "setuptools >= 66.1.0, <= 75.8.0" setuptools
+    '';
+  }));
   ansible_2_16 = python3Packages.toPythonApplication (python3Packages.ansible-core.overridePythonAttrs (oldAttrs: rec {
     version = "2.16.8";
     src = oldAttrs.src.override {
-- 
2.47.1

